Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp

```diff
@@ -4131,3 +4131,118 @@ WebKitFeatureList* webkit_settings_get_development_features(void)
 {
     return webkitFeatureListCreate(WebPreferences::internalDebugFeatures());
 }
+
+/**
+ * webkit_settings_apply_from_key_file:
+ * @settings: a #WebKitSettings
+ * @key_file: a #GKeyFile
+ * @group_name: Name of the group to read from @key_file
+ * @error: return location for error or %NULL to ignore
+ *
+ * Reads the contents of the given @group_name from the given @key_file and apply the value of
+ * each key/value to the corresponding property on the @settings.
+ *
+ * Value types have to match with the corresponding setting property type and the group keys have to
+ * match existing setting property names. If those conditions are not met, the function will return
+ * %FALSE.
+ *
+ * Supported value types are strings (unquoted), booleans (0, 1, true, false) and unsigned integers.
+ *
+ * Returns: %TRUE if the settings were correctly applied or %FALSE on error.
+ *
+ * Since: 2.46
+ */
+gboolean webkit_settings_apply_from_key_file(WebKitSettings* settings, GKeyFile* keyFile, const gchar* groupName, GError** error)
+{
+    g_return_val_if_fail(WEBKIT_IS_SETTINGS(settings), FALSE);
+    g_return_val_if_fail(keyFile, FALSE);
+    g_return_val_if_fail(groupName, FALSE);
+    g_return_val_if_fail(!error || !*error, FALSE);
+
+    GObjectClass* klass = G_OBJECT_GET_CLASS(settings);
+    unsigned totalProperties = 0;
+    GUniquePtr<GParamSpec*> properties(g_object_class_list_properties(klass, &totalProperties));
+
+    if (!g_key_file_has_group(keyFile, groupName)) {
+        g_set_error(error, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_INVALID_VALUE, "The key file has no %s group", groupName);
+        return FALSE;
+    }
+
+    HashMap<String, std::variant<bool, unsigned, String>> values;
+    for (unsigned i = 0; i < totalProperties; i++) {
+        GUniqueOutPtr<GError> lookupError;
+        auto propertyName = makeString(properties.get()[i]->name);
+        if (!g_key_file_has_key(keyFile, groupName, propertyName.utf8().data(), &lookupError.outPtr())) {
```

Comment:
Here we can use properties.get()[i]->name to avoid converting from utf8 and again back to utf8 here.